saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/Xml/NlpXmlReader.md

```diff
@@ -0,0 +1,115 @@
+            <TRAJECTOR id="T4" start="98" end="102" text="a TV"/>
+            <TRAJECTOR id="T5" start="70" end="81" text="wooden desk"/>
+            <TRAJECTOR id="T6" start="86" end="91" text="chair"/>
+            <TRAJECTOR id="T7" start="144" end="156" text="a glass door"/>
+            <TESTPROP id="TP3" first_value="1" second_value="T1" start="62" end="72"/>
+            <MATCH id="inc1" start="70" end="82"/>
+            <MATCH id="inc2" start="143" end="157"/>
+            <MATCH id="p1" start="100" end="101"/>
+            <MATCH id="p2" start="98" end="100"/>
+            <MATCH id="p3" start="100" end="102"/>
+            <MATCH id="o1" start="93" end="99"/>
+            <MATCH id="e1" start="144" end="156"/>
+            <RELATION id="SR4" trajector_id="T4" landmark_id="L3" spatial_indicator_id="S3" general_type="region"
+                      specific_type="RCC8" RCC8_value="EC" FoR="intrinsic"/>
+            <RELATION id="SR5" trajector_id="T5" landmark_id="L4" spatial_indicator_id="S4" general_type="direction"
+                      specific_type="relative" RCC8_value="below" FoR="intrinsic"/>
+            <RELATION id="SR6" trajector_id="T6" landmark_id="L4" spatial_indicator_id="S4" general_type="direction"
+                      specific_type="relative" RCC8_value="below" FoR="intrinsic"/>
+            <RELATION id="SR7" trajector_id="T7" landmark_id="L5" spatial_indicator_id="S5" general_type="region"
+                      specific_type="relative" RCC8_value="DC" FoR="intrinsic"/>
+            <RELATION id="SR8" trajector_id="T7" landmark_id="L6" spatial_indicator_id="S5" general_type="region"
+                      specific_type="relative" RCC8_value="DC" FoR="intrinsic"/>
+        </SENTENCE>
+    </SCENE>
+</SpRL>
+```
+we can read the hierarchy and the relation list by:
+
+```scala
+  val reader = new NlpXmlReader("path_to_the_xml_file.xml", "SCENE", "SENTENCE", null, null)
+  val documentList = reader.getDocuments()
+  val sentencesList = reader.getSentences()
+
+  reader.setPhraseTagName("TRAJECTOR")// set the phrase tag name before reading as phrase list
+  val trajectorList = reader.getPhrases() // reading trajectors as a list of phrases
+  reader.setPhraseTagName("LANDMARK")
+  val landmarkList = reader.getPhrases()
+  reader.setPhraseTagName("SPATIALINDICATOR")
+  val spIndicatorList = reader.getPhrases()
+
+  //the relation contains three arguments which their ids are determined by the specified strings
+  val relationList = reader.getRelations("RELATION", "trajector_id", "spatial_indicator_id", "landmark_id")
+
+  val phraseList = getPhrasesFromSomewhere()// generating the phrases from other sources
+  reader.addPropertiesFromTag("TRAJECTOR", phraseList, XmlMatchings.headwordMatching)
+  reader.addPropertiesFromTag("LANDMARK", phraseList, new XmlPartOfMatching)
+  reader.addPropertiesFromTag("SPATIALINDICATOR", phraseList, new XmlPartOfMatching)
+```
```

Comment:
Although you have the full example here but I think it will be very helpful to have the code inline with your explanations. I think you can keep this block of the code here still.